frontend/src/metabase/query_builder/actions.js

```diff
@@ -1523,50 +1523,55 @@ function getFilterForFK(zoomedObjectId, fk) {
 }
 
 export const FOLLOW_FOREIGN_KEY = "metabase/qb/FOLLOW_FOREIGN_KEY";
-export const followForeignKey = createThunkAction(FOLLOW_FOREIGN_KEY, fk => {
-  return async (dispatch, getState) => {
-    const state = getState();
+export const followForeignKey = createThunkAction(
+  FOLLOW_FOREIGN_KEY,
+  ({ objectId, fk }) => {
+    return async (dispatch, getState) => {
+      const state = getState();
 
-    const card = getCard(state);
-    const queryResult = getFirstQueryResult(state);
-    const zoomedObjectId = getZoomedObjectId(state);
+      const card = getCard(state);
+      const queryResult = getFirstQueryResult(state);
 
-    if (!queryResult || !fk) {
-      return false;
-    }
+      if (!queryResult || !fk) {
+        return false;
+      }
 
-    const newCard = startNewCard("query", card.dataset_query.database);
+      const newCard = startNewCard("query", card.dataset_query.database);
 
-    newCard.dataset_query.query["source-table"] = fk.origin.table.id;
-    newCard.dataset_query.query.filter = getFilterForFK(zoomedObjectId, fk);
+      newCard.dataset_query.query["source-table"] = fk.origin.table.id;
+      newCard.dataset_query.query.filter = getFilterForFK(objectId, fk);
 
-    dispatch(resetRowZoom());
-    dispatch(setCardAndRun(newCard));
-  };
-});
+      dispatch(resetRowZoom());
+      dispatch(setCardAndRun(newCard));
+    };
+  },
+);
```

Comment:
All the diff here is caused by adding an explicit `objectId` argument to `followForeignKey`. It used to select `zoomedObjectId` from redux, but it's not going to be there now if we're drilling by FKs. So now both `followForeignKey` and `loadObjectDetailFKReferences` will expect an explicit `objectId` as an argument